app/models/miq_expression.rb

```diff
@@ -574,180 +574,11 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = _to_sql(@pexp, tz) if @pexp.present?
+    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
```

Comment:
It does :smile: I was hoping for a better diff! It is almost exactly the same code, but with the `to_sql`s removed. I also removed the assignment to `clause` (uses an implicit return) for a few extra :scissors: . That and the composite ones could be simplified a bit. But they are all there!